### ISSUE TITLE
[Bug]: Search for app in App section not working

### ISSUE THREAD

**Issue by tflidd** [nextcloud/server#45558]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

In NC 29.0.1 if I am on the page for the app section and I am searching for an app, e.g. the `deck` app, I get the unified search box with all thinkable results except for apps:

![image](https://github.com/nextcloud/server/assets/8111200/c2c10cf7-f6f9-4096-81b3-4d6696db3bf7)

There is no way of filtering for apps specifically. If I open the search function and I am on the app page, there should be apps-based results first. But now we are searching for everything thinkable except for apps.

### Steps to reproduce

1. Go on apps management page (by default on the discover section)
2. select the search function on the top right corner
3. search for an app (for me deck, text result in a view like in the screen shot, maps shows directly a result for the maps app)

You probably you need a few items in your Nextcloud (Files, messages etc.) that are triggered by the search term.

### Expected behavior

Prioritize apps results in this area. Or have a way to restrict the search for apps.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

29

### Operating system

Other

### PHP engine version

PHP 8.2

### Web server

Nginx

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by solracsf** [comment]
v28 works OK

![image](https://github.com/nextcloud/server/assets/12234510/ffb15387-2857-468f-9155-a10d44318c8d)

**Comment by Neriderc** [comment]
I think there's probably a difference between expectation and implementation.

From what I can tell, the global search will only show apps that are installed. However, when I'm on the app installation page I expect to be able to search apps that are not installed yet. 

Behind the search, the search term seems to filter the apps shown on the page. But it's unclear to me how I'm supposed to search for an app that I know the name of but don't know the category. There's no "all apps" page to filter on, and the global search seems to only show apps that are installed.

**Comment by alexanderdd** [comment]
I wanted to report this, cool that @tflidd already reported it.

These are very common actions, probably the most common actions for app management:
- admin wants to install/uninstall a certain app
- admin wants to check if a certain app is installed, and what version

the process to complete these actions was already not ideal before, and now with NC29 it got even more cumbersome.

Expectation:
1. click on apps in top right menu
2. click on search and type "deck" for example
3. a list appears with all apps that have "deck" in the title, both installed and not installed apps

Current problems:
- the default page when you go to app management is Discover, and the search doesnt work on this page
- problem with the search box being in front of the search results and hiding them -> bad UX (does anybody have a link to the issue?)
- you need to click "Your apps" first, then the search will work -> but the name "your apps" is confusing, it sounds like it only has my installed apps. It should be renamed to "all apps"

**Comment by elibroftw** [comment]
Not just a broken search, but there's three big ass banners that are useless! I appreciate the news, but there's space on the Dashboard for that.

![Image](https://github.com/user-attachments/assets/6e38108b-fe39-4522-88ea-be8355700365)

Here's Google Play Store for reference.

![Image](https://github.com/user-attachments/assets/edfc6ebf-e3f5-4b8b-af4c-e9a77051507a)

**Comment by te-online** [comment]
I could not find a separate issue for this, so I'm adding to this discussion: This might be just me, but I'd prefer there to be no news section in the app store, or an option to disable it. 

It is very distracting.

When I go to the apps "app", I want to 
- install an app, 
- uninstall an app, 
- update apps, 
- search for a specific app or use-case

It's fine to have a landing page to highlight apps that may be interesting or new. But "Nextcloud News" is something that belongs on nextcloud.com and not in the UI of Nextcloud. 

My reasoning is this: I'm using Nextcloud to see less distracting things corporations shove into our faces every day. Distracting me from the task at hand to show me stuff you'd like to show me, but I may not want to see, to me is the definition of a bad UX pattern.

**Comment by Sanookmakmak** [comment]
See also: [https://github.com/nextcloud/fulltextsearch/issues/912](https://github.com/nextcloud/fulltextsearch/issues/912)

And: [https://help.nextcloud.com/t/finde-zeiterfassung-app-nicht/226636/1](https://help.nextcloud.com/t/finde-zeiterfassung-app-nicht/226636/1)

**Comment by pjotrek-b** [comment]
I'm pretty sure, I have the same issue:
I cannot find /any/ app on a fresh nextcloud migration setup to the most recent version (v32.0.1 on Debian12).

The rest of the installation works flawlessly and has no warnings at all in the administration overview.

Any hints greatly appreciated on how to fix this 😄

**Comment by Neriderc** [comment]
> I'm pretty sure, I have the same issue: I cannot find /any/ app on a fresh nextcloud migration setup to the most recent version (v32.0.1 on Debian12).

I'm assuming you can't find apps in the app store (rather than that Nextcloud has no apps installed for you to access).

I have seen this when Nextcloud couldn't access the internet, which is needed to load the app catalogue. In this case you'd still be able to access already installed apps.

Are you using Nextcloud All In One, or did you install manually? These days I would recommend AIO, it does make many things a lot easier to set up.

**Comment by pjotrek-b** [comment]
@Neriderc : Thanks for pointing out "online-connectivity issues of the server" as possible cause.
However: My Nextcloud /can/ acccess the internet. 👍  I've checked - and I also get notification emails (so +1 for proper SMTP config 🍰 )

I can browse, install/uninstall apps - no problem: but **I cannot search** them. 😞 

Not sure if it's related, but when enabling the "AppAPI" app, I get the following warning:

> AppAPI deploy daemon
> AppAPI default deploy daemon is not set. Please register a default deploy daemon in the settings to install External Apps (Ex-Apps).

**Comment by pjotrek-b** [comment]
Forgot to mention, regarding "AIO" or not:
This is a migrated installation:
  - from nextcloud v22 / ubuntu16.04
  - to nextcloud v32 / debian12

So the code is "copy-paste from old server to new one" - and then "occ update" until v32.

**Comment by pjotrek-b** [comment]
UPDATE: I fixed it! 🥳 
I'm not yet fully sure how, but:

* This nextcloud-apache instance is running behind a reverse-proxy
* The ServerName of the **proxied** nextcloud-apache /was/ the internal hostname (eg "cloud.local")
* I've now changed the ServerName of that instance to the FQDN real-world hostname (eg "cloud.example.com")

So @Neriderc, you may have been right all along, and there was a "fishy-servername" config which may have prevented the proxied nextcloud to properly "access internet and itself" after all? 😉 😄

**Comment by szaimen** [comment]
@susnux should we disable the search feature in the discover section and add a popup that says that you need to navigate to a different section to use search?

Also cc @nextcloud/designers

**Comment by marcoambrosini** [comment]
This conversation is a bit confusing because there are different issues. The original bug seems to be solved? 

Regarding 
> @susnux should we disable the search feature in the discover section and add a popup that says that you need to navigate to a different section to use search?

I don't think so, we should allow searching in the discover section and whenever a search operation is in progress, replace the entire view with search results from all categories

**Comment by susnux** [comment]
> I don't think so, we should allow searching in the discover section and whenever a search operation is in progress, replace the entire view with search results from all categories

Agree and this is on my schedule but likely not until NC33 things are done.

**Comment by susnux** [comment]
Fixed in #57290
